### PR TITLE
Widget perf polish

### PR DESCRIPTION
**Pull request by jacob314**
Switch to the new event based approach for widget perf events that has landed in package:flutter
With this change you can test the behavior using regular tip of trunk flutter.

Add UI displaying performance tips for common flutter performance problems.
Hide track widget repaint option from the ui by default to focus on just repaints.
Add checkboxes in the perf tab turning on and off the performance overlay and repaint rainbows.

Turn off animations on highlighted widget names as they increased cpu usage significantly.
screen shot 2018-10-30 at 9 30 55 am

### CODE REVIEW

File: src/io/flutter/perf/WidgetPerfLinter.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      "Performance optimizations when using AnimatedBuilder",
+      "https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/AnimatedBuilder-class.html#performance-optimizations",
+      matchParent("AnimatedBuilder"),
+      1,
+      50,
+      20,
+      AllIcons.Actions.IntentionBulb
+    ));
+
+
```
Comment (stevemessick) [inline src/io/flutter/perf/WidgetPerfLinter.java]: Extra line.

File: src/io/flutter/perf/WidgetPerfLinter.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      "perf_diagnosis_demo/lib/clock_demo.dart",
+      "Performance considerations of StatefulWidget",
+      "https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/StatefulWidget-class.html#performance-considerations",
+      matchParent("StatefulWidget"),
+      4, // Only relevant if the build method is somewhat large.
+      50,
+      20,
+      AllIcons.Actions.IntentionBulb
+    ));
+    tips.add(new PerfTipRule(
```
Comment (stevemessick) [inline src/io/flutter/perf/WidgetPerfLinter.java]: Missing blank line (for consistency).

File: src/io/flutter/perf/FlutterWidgetPerf.java [hunk 3]
```diff
@@ -149,82 +251,81 @@ private void performRequest(TextEditor[] fileEditors) {
...
-          uiAnimationTimer.stop();
+        final TextRange range = fileLocationMapper.getIdentifierRange(location.line, location.column);
+        if (range == null) {
+          continue;
         }
+        fileStats.add(
+          range,
+          new SummaryStats(
+            kind,
+            new SlidingWindowStatsSummary(entry, lastStartTime, location), // XXX need to handle timeout better.
```
[the remark this answers is not in the document]
Comment (jacob314) [reply src/io/flutter/perf/FlutterWidgetPerf.java]: doh. XXX comments are placeholders to remind me to remove the comments before commit.
we now handle timeouts well.

File: src/io/flutter/perf/WidgetPerfLinter.java (hunk 1 above)
Comment (jacob314) [reply src/io/flutter/perf/WidgetPerfLinter.java]: done.

File: src/io/flutter/perf/WidgetPerfLinter.java (hunk 2 above)
Comment (jacob314) [reply src/io/flutter/perf/WidgetPerfLinter.java]: done

**Pull request merged**